### PR TITLE
Fix transformations and accuracy in text recognition

### PR DESCRIPTION
**Pull request by morkovka1337**
A small fix for text recognition. 
1. Fixes a bug when resize transform was applied after filtering images of different size in batch and therefore images were dropped from batch even when resized to fixed shape.
2. Fixes a small bug in training script which lead to wrong accuracy calculation during training.

### CODE REVIEW

**Comment by morkovka1337** [comment]
Run tests

**Comment by morkovka1337** [comment]
run tests

File: misc/pytorch_toolkit/text_recognition/text_recognition/data/utils.py [hunk 1]
```diff
@@ -441,21 +441,26 @@ def get_num_lines_in_file(path):
...
-    batch = [img_text for img_text in batch if img_text['img'].shape == size]
     # sort by the length of text
     # the purpose of the sort is to put the longest text on the first place
     # to get correct size of the tensor in the texts2tensor function
     batch.sort(key=lambda img_text: len(img_text['text'].split()), reverse=True)
 
+    imgs = [item['img'] for item in batch]
+    if batch_transform:
+        imgs = batch_transform(imgs)
+    for i, item in enumerate(batch):
```
Comment (Ilya-Krylov) [inline misc/pytorch_toolkit/text_recognition/text_recognition/data/utils.py:452]: is it kind of redundant activity if batch_transform is False?

Comment (Ilya-Krylov) [reply misc/pytorch_toolkit/text_recognition/text_recognition/data/utils.py:452]: together with line 449

File: misc/pytorch_toolkit/text_recognition/text_recognition/utils/trainer.py [hunk 2]
```diff
@@ -145,6 +145,7 @@ def __init__(self, work_dir, config):
             'CTCLossZeroInf', False)) if self.loss_type == 'CTC' else None
         self.out_size = len(self.vocab) + 1 if self.loss_type == 'CTC' else len(self.vocab)
         self.model = TextRecognitionModel(config.get('backbone_config'), self.out_size, config.get('head', {}))
+        print(self.model)
```
Comment (Ilya-Krylov) [inline misc/pytorch_toolkit/text_recognition/text_recognition/utils/trainer.py:148]: is it needed?

**Comment by Ilya-Krylov** [comment]
@morkovka1337 are you going to resolve comments above?

**Comment by morkovka1337** [comment]
Did not see them. Yes.

File: misc/pytorch_toolkit/text_recognition/text_recognition/data/utils.py (hunk 1 above)
Comment (morkovka1337) [reply misc/pytorch_toolkit/text_recognition/text_recognition/data/utils.py]: Moved this code under if condition.

File: misc/pytorch_toolkit/text_recognition/text_recognition/utils/trainer.py (hunk 2 above)
Comment (morkovka1337) [reply misc/pytorch_toolkit/text_recognition/text_recognition/utils/trainer.py:148]: This is a good log message about a model.

**Comment by morkovka1337** [comment]
run tests

**Comment by morkovka1337** [comment]
run tests

**Review by Ilya-Krylov — APPROVED** [review]
